src/all/batoto/src/eu/kanade/tachiyomi/extension/all/batoto/BatoToUrlActivity.kt

```diff
@@ -42,8 +42,25 @@ class BatoToUrlActivity : Activity() {
 
     private fun fromBatoTo(pathSegments: MutableList<String>): String? {
         return if (pathSegments.size >= 2) {
-            val id = pathSegments[1]
-            "ID:$id"
+            val path = pathSegments[1] as java.lang.String?
```

Comment:
wouldn't `pathSegment[1]` already be `String`? so why cast

Reply:
Because if I remove the cast or cast it to String?, it uses Intrinsics class:
```java
if (index == -1) {
    var10000 = path;
} else {
    var10000 = path.substring(0, index);
    Intrinsics.checkNotNullExpressionValue(var10000, "this as java.lang.String…ing(startIndex, endIndex)");
}

String id = var10000;
```
If I cast it to java.lang.String?, it doesn't use Intrinsics class:
```java
Object id = index == -1 ? path : path.substring(0, index);
```